pkg/controller/knativeserving/common/images_test.go

```diff
@@ -0,0 +1,117 @@
+package common
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	appsv1 "k8s.io/api/apps/v1"
+)
+
+type updateImageTest struct {
+	name          string
+	in            string
+	registry      string
+	tag           string
+	expected      string
+	expectedError bool
+}
+
+var updateImageTests = []updateImageTest{
+	{
+		name:     "NewRegistryAndTagFromDigest",
+		in:       "gcr.io/knative-releases/github.com/knative/serving/cmd/queue@sha256:1e40c99ff5977daa2d69873fff604c6d09651af1f9ff15aadf8849b3ee77ab45",
+		registry: "new-registry.io/test/path",
+		tag:      "new-tag",
+		expected: "new-registry.io/test/path/queue:new-tag",
+	},
```

Comment:
Another challenge just occurred to me: RHT currently renames its images with a `knative-serving-` prefix, so `queue` becomes `knative-serving-queue`. Hence, our image name will be `quay.io/openshift-knative/knative-serving-queue:v0.6.0`. Will the current logic stick a `/` in there? If so, can we just expect the separator to be in the name, e.g `quay.io/openshift-knative/knative-serving-` and `new-registry.io/test/path/`?